src/main/scala/io/iohk/ethereum/blockchain/sync/regular/ExecutionSync.scala

```diff
@@ -0,0 +1,142 @@
+package io.iohk.ethereum.blockchain.sync.regular
+
+import akka.actor.{Actor, ActorLogging, ActorRef, AllForOneStrategy, Props, Scheduler, SupervisorStrategy}
+import io.iohk.ethereum.blockchain.sync.{Blacklist, SyncProtocol}
+import io.iohk.ethereum.blockchain.sync.regular.RegularSync.{NewCheckpoint, ProgressState}
+import io.iohk.ethereum.consensus.validators.BlockValidator
+import io.iohk.ethereum.db.storage.StateStorage
+import io.iohk.ethereum.domain.{Block, Blockchain, BlockchainReader}
+import io.iohk.ethereum.ledger.{BlockImport, BranchResolution}
+import io.iohk.ethereum.network.Peer
+import io.iohk.ethereum.network.PeerEventBusActor.SubscriptionClassifier.MessageClassifier
+import io.iohk.ethereum.network.p2p.messages.Codes
+import io.iohk.ethereum.utils.ByteStringUtils
+import io.iohk.ethereum.utils.Config.SyncConfig
+import io.iohk.ethereum.network.PeerEventBusActor.{PeerSelector, Subscribe}
+
+class ExecutionSync(
+    fetcherService: FetcherService,
+    blockImporterService: BlockImporterService,
+    peersClient: ActorRef,
+    etcPeerManager: ActorRef,
+    peerEventBus: ActorRef,
+    blockImport: BlockImport,
+    blockchain: Blockchain,
+    blockchainReader: BlockchainReader,
+    stateStorage: StateStorage,
+    branchResolution: BranchResolution,
+    blockValidator: BlockValidator,
+    blacklist: Blacklist,
+    syncConfig: SyncConfig,
+    ommersPool: ActorRef,
+    pendingTransactionsManager: ActorRef,
+    scheduler: Scheduler
+) extends Actor
+    with ActorLogging {
+
+  val broadcaster: ActorRef = context.actorOf(
+    BlockBroadcasterActor
+      .props(new BlockBroadcast(etcPeerManager), peerEventBus, etcPeerManager, blacklist, syncConfig, scheduler),
+    "block-broadcaster"
+  )
+
+  override def receive: Receive = running(
+    ProgressState(initialBlock = 0, currentBlock = 0, bestKnownNetworkBlock = 0)
+  )
+
+  def running(progressState: ProgressState): Receive = {
+
+    case SyncProtocol.Start =>
+      log.info("Starting regular sync")
+      peerEventBus ! Subscribe(
+        MessageClassifier(
+          Set(Codes.NewBlockCode, Codes.NewBlockHashesCode, Codes.BlockHeadersCode),
+          PeerSelector.AllPeers
+        )
+      )
+      val listOfPeersToFetchFromWithBestBlock: List[(Peer, Int)] = ???
+      val startFrom: BigInt = blockchainReader.getBestBlockNumber()
+      listOfPeersToFetchFromWithBestBlock.map(peerWithBlockNumber =>
```

Comment:
style remark (which is of course personal): I would have gone with: 
```
listOfPeersToFetchFromWithBestBlock.map { case (peerId, peerBestBlockNumber) =>
  fetcherService.fetchBlocksUntil(peerId, Left(startFrom), Left(peerBestBlockNumber))
}
```

I tend to find it easier to read a case(a, b) than a tuple element acces like data._1, even though in this case the name is pretty straightforward